app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt

```diff
@@ -83,11 +114,26 @@ object NetworkModule : InjectionModule {
         }
     }
 
-    private fun Scope.getClient(tag: String, interceptor: Interceptor? = null): OkHttpClient {
+    private fun Scope.getClient(tag: String, resources: Resources, interceptor: Interceptor? = null): OkHttpClient {
         return OkHttpClient.Builder()
             .readTimeout(DEFAULT_CONNECT_TIMEOUT_SECONDS, TimeUnit.SECONDS)
             .connectTimeout(DEFAULT_READ_TIMEOUT_SECONDS, TimeUnit.SECONDS)
             .apply {
+                val appFeatureFlags: AppFeatureFlags = get()
+                if (appFeatureFlags.isSslPinningEnabled) {
+                    try {
+                        createCertificate(resources.openRawResource(R.raw.cert))?.let { sslContext ->
```

Comment:
I suggest to extract all this pinning logic to a separate entity callled smth like "SslCertificatePinner" and then pass okhttp
it's not Module responsiblity to do such things.

Reply:
also, do we need to pin SSL to all the Retrofit instances, or just FeeRelayer?